src/daos_vol_link.c

```diff
@@ -1911,6 +1911,7 @@ H5_daos_link_create_task(tse_task_t *task)
 	    if(H5_daos_create_task(H5_daos_metatask_autocomplete, 1, &dep_task,  NULL, NULL, NULL, &metatask) < 0)
 		D_DONE_ERROR(H5E_LINK, H5E_CANTINIT, FAIL, "can't create metatask for link create");
 
+	    tse_task_register_deps(task, 1, &metatask);
```

Comment:
This change looks reasonable to me, but @fortnern might want to comment on it.

Reply:
This seems strange, how does this task get completed now? Does DAOS complete the task automatically if it gets dependencies in the body function? I vaguely remember a weird pattern like this in the DAOS tests.

I made a mistake when I signed off on the original change. Instead of H5_daos_metatask_autocomplete this should use H5_daos_metatask_autocomp_other, with task passed as the private data. This explicitly has the metatask complete the link create task when it executes.

Reply:
the previous way was still not working and causing that link test to fail.
let me try with your suggested change instead to see if it helps.

Reply:
looks like H5_daos_metatask_autocomp_other works well. ill update that.